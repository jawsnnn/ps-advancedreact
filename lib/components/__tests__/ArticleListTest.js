const React = require("react");
const renderer = require("react-test-renderer");
const ArticleList = require("../ArticleList");

describe("Article list tests", () => {

    const testProps = {
        articles: {
            a: {id: 1},
            b: {id: 2}
        },
        articleActions: {
            lookupAuthor: jest.fn(()=>({}))
        }
    };

    it("renders correctly", () => { 
        const tree = renderer.create(
            <ArticleList 
                {...testProps}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toBe(2);
    });
});