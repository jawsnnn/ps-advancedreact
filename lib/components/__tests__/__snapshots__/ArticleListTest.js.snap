// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article list tests renders correctly 1`] = `
<div>
  <div>
    <div />
    <div />
    <a
      href={undefined}
    >
      <div>
         
      </div>
    </a>
    <div />
  </div>
  <div>
    <div />
    <div />
    <a
      href={undefined}
    >
      <div>
         
      </div>
    </a>
    <div />
  </div>
</div>
`;
